projects/ad9208/main.c

```diff
@@ -0,0 +1,480 @@
+		.pulse_gen_mode = 7, /* HMC7044_PULSE_GEN_CONT_PULSE */
+		.in_buf_mode = {0x07, 0, 0, 0, 0x03},
+		.gpi_ctrl = {0x00, 0x00, 0x00, 0x00},
+		.gpo_ctrl = {0x1f, 0x2b, 0x00, 0x00},
+		.num_channels = 8,
+		.channels = chan_spec,
+	};
+
+	struct spi_init_param ad9208_0_spi_param = {
+		.id = SPI_0_DEVICE_ID,
+		.max_speed_hz = 10000000,
+		.mode = SPI_MODE_3,
+		.chip_select = SPI_AD9208_0_CS,
+		.flags = 0,
+	};
+
+	struct spi_init_param ad9208_1_spi_param = {
+		.id = SPI_DEVICE_ID,
+		.max_speed_hz = 10000000,
+		.mode = SPI_MODE_3,
+		.chip_select = SPI_AD9208_CS,
+		.flags = 0,
+	};
+
+	struct ad9208_ddc ad9208_0_ddc_init[1] = {
+		{
+			.decimation = 2,
+			.nco_mode = AD9208_NCO_MODE_VIF,
+			.carrier_freq_hz = 70000000,
+			.po = 0,
+			.gain_db = true,
+		},
+	};
+
+	struct ad9208_ddc ad9208_1_ddc_init[1] = {
+		{
+			.decimation = 2,
+			.nco_mode = AD9208_NCO_MODE_VIF,
+			.carrier_freq_hz = 70000000,
+			.po = 0,
+			.gain_db = true,
+		},
+	};
+
+	jesd_param_t jesd_param_init = {
+			.jesd_F = 1, /* octets per frame */
+			.jesd_K = 32, /* frames per multiframe */
+			.jesd_HD = false, /* high density */
+			.jesd_N = 16, /* converter resolution */
+			.jesd_NP = 16, /* bits per sample */
+			.jesd_M = 2, /* converters per device */
+			.jesd_CS = 0, /* control bits per sample */
+			.jesd_L = 8, /* lanes per device */
+	};
+
+	struct adxcvr_init rx_0_adxcvr_init = {
+		.name = "rx_0_adxcvr",
+		.base = RX_0_XCVR_BASEADDR,
+		.sys_clk_sel = 3,
+		.out_clk_sel = 3,
+		.cpll_enable = 0,
+		.lpm_enable = 1,
+		/* LaneRate = (M/L)*NP*(10/8)*DataRate */
+		.lane_rate_khz = 15000000,
+		/* FMC_REFCLK0, output 8 of HMC 7044 */
+		.ref_rate_khz = 750000,
+	};
+
+	struct adxcvr_init rx_1_adxcvr_init = {
+		.name = "rx_1_adxcvr",
+		.base = RX_1_XCVR_BASEADDR,
+		.sys_clk_sel = 3,
+		.out_clk_sel = 3,
+		.cpll_enable = 0,
+		.lpm_enable = 1,
+		/* LaneRate = (M/L)*NP*(10/8)*DataRate */
+		.lane_rate_khz = 15000000,
+		/* FMC_REFCLK0, output 8 of HMC 7044 */
+		.ref_rate_khz = 750000,
+	};
+
+	struct jesd204_rx_init rx_0_jesd_init = {
+		.name = "rx_0_jesd",
+		.base = RX_0_JESD_BASEADDR,
+		.octets_per_frame = 1,
+		.frames_per_multiframe = 32,
+		.subclass = 1,
+		/* (lane_clk_khz / 40) output 9 of HMC 7044*/
+		.device_clk_khz = 375000,
+		/* LaneRate = (M/L)*NP*(10/8)*DataRate */
+		.lane_clk_khz = 15000000,
+	};
+
+	struct jesd204_rx_init rx_1_jesd_init = {
+		.name = "rx_1_jesd",
+		.base = RX_1_JESD_BASEADDR,
+		.octets_per_frame = 1,
+		.frames_per_multiframe = 32,
+		.subclass = 1,
+		/* (lane_clk_khz / 40) output 9 of HMC 7044*/
+		.device_clk_khz = 375000,
+		/* LaneRate = (M/L)*NP*(10/8)*DataRate */
+		.lane_clk_khz = 15000000,
+	};
+
+	struct axi_adc_init rx_0_adc_init = {
+		"rx_0_adc",
+		RX_0_CORE_BASEADDR,
+		2,
+	};
+
+	struct axi_adc_init rx_1_adc_init = {
+		"rx_1_adc",
+		RX_1_CORE_BASEADDR,
+		2,
+	};
+
+	struct axi_dmac_init rx_dmac_init = {
+		"rx_dmac",
+		RX_DMA_BASEADDR,
+		DMA_DEV_TO_MEM,
+		0,
+	};
```

Comment:
Probably we should use explicit names, for structure elements, when initializing :
.name  = "rx_dmac"